src/dnddiceroller/Die.java

```diff
@@ -9,4 +11,34 @@
  */
 public class Die {
   
+    //fileds
+    final private int sides;
+    private int value;
+
+/**
+ * the constructor performs initial roll of the die.
+ * @param numSides is the number of sides for this die.
+ */
+    public Die(int numSides){
+               
+        sides = numSides;
+        roll();
+}
+     
```

Comment:
@srtinkey , neither did I until I spent this past week studying our text book and this was suggested in a piece of their code, so I tried it out.